src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -93,6 +116,66 @@ PluginResponse handlePluginsRequest(PluginRequest pluginRequest) {
         return pluginResponse;
     }
 
+    NamedWriteableRegistryResponse handleNamedWriteableRegistryRequest(NamedWriteableRegistryRequest request) {
+
+        logger.info("Registering Named Writeable Registry Request recieved from OpenSearch.");
+
+        // iterate through Extensions's named writeables and add to extension entries
+        Map<String, String> extensionEntries = new HashMap<>();
+        for (NamedWriteableRegistry.Entry entry : this.namedWriteables) {
+            extensionEntries.put(entry.name, entry.categoryClass.getName());
+        }
+        NamedWriteableRegistryResponse namedWriteableRegistryResponse = new NamedWriteableRegistryResponse(extensionEntries);
+        return namedWriteableRegistryResponse;
+    }
+
+    <C extends NamedWriteable> NamedWriteableRegistryParseResponse handleNamedWriteableRegistryParseRequest(
+        NamedWriteableRegistryParseRequest request
+    ) throws IOException {
+
+        logger.info("Registering Named Writeable Registry Parse request from OpenSearch");
+        boolean status = false;
+
+        // extract data from request and procress fully qualified category class name into class instance
+        try {
```

Comment:
Do we need a lot of nested `try`'s?

Reply:
I went ahead and removed the nested try's from the response handlers. As per Dan's suggestions, I will now handle Class object creation within the request/response constructors